test/Microsoft.AspNetCore.Routing.Tests/Template/TemplateMatcherTests.cs

```diff
@@ -991,6 +991,107 @@ public void TryMatch_MultipleOptionalParameters()
             Assert.False(values.ContainsKey("id"));
         }
 
+        [Theory]
+        [InlineData("///")]
+        [InlineData("/a//")]
+        [InlineData("/a/b//")]
+        [InlineData("//b//")]
+        [InlineData("///c")]
+        [InlineData("///c/")]
+        public void TryMatch_MultipleOptionalParameters_WithEmptyIntermediateSegmentsDoesNotMatch(string url)
```

Comment:
Added tests for attribute routing, I think we don't need tests for route as it's pretty straightforward what we do in the RouteBase implementation. If you think otherwise, I think the right thing would be to create a bunch of tests that all route implementations can reuse instead of having the same kind of tests duplicated across the code base, but I think we are covered for this bug